Automattic/wp-calypso
# Add "Switch site" page

Part of https://github.com/Automattic/wp-calypso/issues/88229
Required by https://github.com/Automattic/wp-calypso/pull/88913

## Proposed Changes

Adds a new `/switch-site` page which displays the users sites and then after the user selects one, redirects to that site with the given `route` param (e.g. `/switch-site?route=/wp-admin/edit.php` or `/switch-site?route=/posts/:site`). 

<img width="500" alt="Screenshot 2024-04-05 at 16 28 23" src="https://github.com/Automattic/wp-calypso/assets/1233880/be81968f-35e5-4de8-a9ef-721e99a8d0a8">

### Issues & potential issues
For any given page, the user may or may not have access to that page on some or all of their other sites. Due to plugins, we can't know this in advance for atomic sites. _Maybe_ something could be worked out for simple sites but I'm not sure it's worth the effort. This means that all sites are listed in the site selector and we rely upon the site/plugin to handle this reasonably. Here's what it looks like for the plugins page:



https://github.com/Automattic/wp-calypso/assets/1233880/f8e20979-7631-4d00-8592-58ab8be24524




## Testing Instructions

- Use the Calypso live link below
- Go to `/switch-site?route=<PATH>` and try different routes (e.g. /wp-admin/edit.php, or /posts/:site)
- You should see a list of your sites 
- Choose a site
- You should get sent to the given route for that site
